lib/fs.js

```diff
@@ -232,6 +248,8 @@ fs.accessSync = function(path, mode) {
 };
 
 fs.exists = function(path, callback) {
+  if (handleError((path = getPathFromURL(path)), callback))
```

Comment:
I disagree here. Note that the `nullCheck()` method check implements the same basic behavior (throwing or doing a callback with an error). Given that the `getPathFromURL()` method is not performing any actual file checks and instead throws based on syntax validation of the input.